components/compositing/build.rs

```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+extern crate toml;
+
+use std::env;
+use std::fs::File;
+use std::io::{Read, Write};
+
+const WEBRENDER_REVISION_TEMPLATE: &'static str =
+"/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/// auto-generated by cargo. do not manually modify.
+pub const REVISION: &'static str = ";
+
+fn main() {
+    let current_path = env::current_dir().expect("Failed to access path to lockfile");
+    let lockfile_path = current_path.join("..").join("..").join("Cargo.lock");
+    let revision_file_path = current_path.join("webrender_revision.rs");
```

Comment:
Rather than creating a file in the source directory, we should use the OUT_DIR environment variable and then include the file like [this](https://github.com/servo/servo/blob/master/components/script/dom/mod.rs#L208-L210).

Reply:
that sounds legit.